aten/src/ATen/native/quantized/QTensor.cpp

```diff
@@ -40,8 +40,13 @@ Tensor quantize_per_channel_cpu(
     const Tensor& zero_points,
     int64_t axis,
     ScalarType dtype) {
-  auto quantizer =
-      make_per_channel_affine_quantizer(scales, zero_points, axis, dtype);
+  QuantizerPtr quantizer;
+  if (zero_points.scalar_type() == kFloat) {
+    quantizer = make_per_row_float_qparams_quantizer(scales, zero_points, axis, dtype);
+  } else {
+    quantizer =
+        make_per_channel_affine_quantizer(scales, zero_points, axis, dtype);
```

Comment:
Can we overload the make_per_channel_affine_quantizer function here to handle different dtypes for scale and zero-point?